### ISSUE TITLE
Pasting every emoticon into a pad on a single line makes the pad run weird and import is somewhat broken.

### ISSUE THREAD

**Issue by JohnMcLear** [ether/etherpad-lite#3738]
demo file: https://github.com/ether/etherpad-lite/pull/3737/files#diff-9511b7171d0f22a41e2d6000d4c8934a

There are a few problems after copy/pasting the content from the emoticons HTML.

- [x] Character can jump around. - Caused by a bad commit that was reverted
- [x] Render feels clunky and slow. - Caused by a bad commit that was reverted
- [ ] **Despite it being a valid HTML file import fails using the UI.** -- Note that import works using setHTML API

Steps to validate backend -- Edit:  Backend is now fixed (PR waiting to be merged)

1. Checkout JohnMcLear:backendTestForUTF8Encoding
1. ``cd src``
1. ``npm run tests``

Steps to validate frontend
1. Checkout JohnMcLear:backendTestForUTF8Encoding
1. Attempt to import the emoticons.html file (/tests/backend/specs/emoticons.html)

Etherpad sails silently to setHTML (on import -- provides no response) but in console.warn you get ``[WARN] ImportHtml - HTML was not properly formed { inspect:[Function: inspect] }``

Testing the file through the browser import feature also fails as expected.

**Comment by JohnMcLear** [comment]
Ugh, okay so this uses ``contentcollector.js`` which is client and server side code and it's super legacy.  

Thankfully it's only 800 lines so pinpointing where it's failing might not be too painful BUT it does mean that contentcollector.js (again used both client and server) has some fundamental flaws for handling HTML which is a good thing we found this because this could be the cause of some other issues surrounding changeset reps et al.

TLDR; A big core part of the client and server side code doesn't handle properly formed HTML properly.  This needs fixing, ASAP!

**Comment by JohnMcLear** [comment]
@muxator afaik we can close this now?
